packages/types/definitions.ts

```diff
@@ -0,0 +1,514 @@
+      >;
+    }
+  | {
+      /**
+       * @title Engine
+       */
+      engine: 'socketio';
+      /**
+       * @title Scenario flow
+       */
+      flow: Array<
+        | SocketIoFlow
+        | ({
+            loop: Array<SocketIoFlow>;
+            whileTrue?: string;
+          } & (FixedLoop | DynamicLoop))
+      >;
+    }
+  | {
+      /**
+       * @title Engine
+       */
+      engine: 'playwright';
+      /**
+       * @title Scenario flow
+       */
+      flow: Array<
+        | any
+        | ({
+            loop: Array<any>;
+            whileTrue?: string;
+          } & (FixedLoop | DynamicLoop))
+      >;
+    }
+);
+
+export type FixedLoop = {
+  /**
+   * Exact number of times to loop through the scenario.
+   * @title Count
+   */
+  count: number;
+};
+
+export type DynamicLoop = {
+  /**
+   * Array of values to loop through.
+   * Alternatively, a name of the variable containing the array.
+   * @title Over
+   */
+  over: string | Array<string>;
+};
+
+export type BaseFlow =
+  | {
+      /**
+       * Print given message to the console.
+       * @title Log
+       */
+      log: string;
+    }
+  | {
+      /**
+       * Pause virual user for the given duration (in seconds).
+       * @title Think
+       */
+      think: number;
+    }
+  | {
+      /**
+       * Function name to run.
+       * @title Function
+       */
+      function: string;
+    };
+
+export type HttpFlow =
+  | BaseFlow
+  | {
+      /**
+       * @title Perform a GET request
+       */
+      get: DefaultHttpRequest;
+    }
+  | {
+      /**
+       * @title Perform a POST request
+       */
+      post: DefaultHttpRequest | HttpRequestWithBody;
+    }
+  | {
+      /**
+       * @title Perform a PUT request
+       */
+      put: DefaultHttpRequest | HttpRequestWithBody;
+    }
+  | {
+      /**
+       * @title Perform a PATCH request
+       */
+      patch: DefaultHttpRequest | HttpRequestWithBody;
+    }
+  | {
+      /**
+       * @title Perform a DELETE request
+       */
+      delete: DefaultHttpRequest | HttpRequestWithBody;
+    };
+
+export type WebSocketFlow =
+  | BaseFlow
+  | {
+      connect:
+        | string
+        | {
+            function: string;
+          }
+        | {
+            target: string;
+            proxy?: {
+              url: string;
+            };
+          };
+    }
+  | {
+      send: string | object;
+    };
+
+export type SocketIoFlow =
+  | BaseFlow
+  | {
+      emit: {
+        channel: string;
+        data: string;
+        namespace?: string;
+        response?: {
+          channel: string;
+          data: string;
+        };
+        acknowledge?: {
+          data?: string;
+          match?: {
+            json: any;
+            value: string;
+          };
+        };
+      };
+    };
+
+export type DefaultHttpRequest = {
+  /**
+   * @title URL
+   */
+  url: string;
+  /**
+   * @title Headers
+   */
+  headers?: object;
+  /**
+   * @title Cookie
+   */
+  cookie?: string;
```

Comment:
`cookie` isn't a `string` I believe:
https://www.artillery.io/docs/reference/engines/http#cookies

Seems to be key:value pair (where value is a string)

Reply:
Thanks for spotting this! Updated.